modules/web/src/app/shared/validators/encryption-key.validator.ts

```diff
@@ -0,0 +1,39 @@
+// Copyright 2025 The Kubermatic Kubernetes Platform contributors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {AbstractControl, ValidationErrors, Validator} from '@angular/forms';
+
+export class EncryptionKeyValidator implements Validator {
+  validate(control: AbstractControl): ValidationErrors | null {
+    const value = control.value;
+    if (!value) {
+      return null;
+    }
+    // 32-byte key must be exactly 44 base64 characters (43 chars + 1 padding)
+    const base64Regex = /^[A-Za-z0-9+/]{43}=$/;
```

Comment:
Hi, @adoi 

Regex Adds Strict Validation to check
  1. 32-byte keys must encode to exactly 44 base64 characters
  2. The format is specifically 43 base64 characters + 1 padding character

so, it string "Rand" generated or User provided should always contain "Padding" character to pass validation check

Let me know if any concerns